packages/pyramid_lint/lib/src/lints/flutter/dispose_controllers.dart

```diff
@@ -224,20 +224,33 @@ Iterable<String> _getDisposeStatementTargetNames(
   NodeList<Statement> statements,
 ) {
   return statements.expressionStatements
-      .map((e) => e.expression)
-      .whereType<MethodInvocation>()
       .map(_getTargetNameOfDisposeMethodInvocation)
       .nonNulls;
 }
 
 String? _getTargetNameOfDisposeMethodInvocation(
-  MethodInvocation invocation,
+  ExpressionStatement expressionStatement,
 ) {
-  final target = invocation.target;
-  if (target is! SimpleIdentifier) return null;
+  if (expressionStatement.expression is CascadeExpression) {
+    final cascadeExpression =
+        expressionStatement.expression as CascadeExpression;
```

Comment:
Shall we simplify this with `if (expressionStatement.expression case final CascadeExpression cascadeExpression)`?